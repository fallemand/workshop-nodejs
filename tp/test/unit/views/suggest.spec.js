const React = require('react');
const renderer = require('react-test-renderer');
const Suggest = require('../../../views/components/Suggest');

describe('Suggest (Snapshot)', () => {
  const component = renderer.create(<Suggest query="iph" />);
  const response = {
    results: ['iphone', 'iphone 6']
  };
  window.fetch = jest.fn().mockImplementation(
    () => Promise.resolve({json: () => Promise.resolve(response)})
  );

  afterEach(() => {
    const instance = component.getInstance();
    instance.setState({
      results: []
    });
  });

  it('Suggest renders without results', () => {
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('Suggest renders with results [change state]', () => {
    const instance = component.getInstance();
    instance.setState({
      results: ['iphone', 'iphone2']
    });
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('Suggest renders with results [componentwillReceiveProps]', () => {
    const instance = component.getInstance();
    expect.assertions(1);
    instance.componentWillReceiveProps({query: 'iph'}).then(() => {
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    });

  });


});
