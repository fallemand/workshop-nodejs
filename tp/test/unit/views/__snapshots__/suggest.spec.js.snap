// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggest (Snapshot) Suggest renders with results [change state] 1`] = `
<div>
  <ul
    className="suggest"
  >
    <li
      className="suggest__item"
    >
       
      <a
        href="/app/search?q=iphone"
      >
        iphone
      </a>
    </li>
    <li
      className="suggest__item"
    >
       
      <a
        href="/app/search?q=iphone2"
      >
        iphone2
      </a>
    </li>
  </ul>
</div>
`;

exports[`Suggest (Snapshot) Suggest renders with results [componentwillReceiveProps] 1`] = `null`;

exports[`Suggest (Snapshot) Suggest renders with results 22 [componentwillReceiveProps] 1`] = `
<div>
  <ul
    className="suggest"
  >
    <li
      className="suggest__item"
    >
       
      <a
        href="/app/search?q=iphone"
      >
        iphone
      </a>
    </li>
    <li
      className="suggest__item"
    >
       
      <a
        href="/app/search?q=iphone2"
      >
        iphone2
      </a>
    </li>
  </ul>
</div>
`;

exports[`Suggest (Snapshot) Suggest renders without results 1`] = `<div />`;
